### PR TITLE
backupccl: restore role options during RESTORE SYSTEM USERS

### LINKED ISSUE #94107: restore: RESTORE SYSTEM USERS does not restore role_options
**Describe the problem**

`RESTORE SYSTEM USERS` does not restore the system.role_options table.

Please describe the issue you observed, and any steps we can take to reproduce it:

**To Reproduce**

- `CREATE USER u WITH PASSWORD 'pass' VALID UNTIL '2050-01-01';`
- `BACKUP TO 'nodelocal://0/foo/1';`
- In a new cluster, `RESTORE SYSTEM USERS FROM 'nodelocal://0/foo/1';`

User `u` does not have a password expiration. Check with `SELECT * from system.role_options WHERE username = 'u';`

**Expected behavior**
The user should have all its role options

**Additional context**
The original PR had a comment that stated that role_options should be handled: https://github.com/cockroachdb/cockroach/pull/71542#discussion_r749434407. But the PR got merged without addressing it.

Jira issue: CRDB-22696

### PR DESCRIPTION
**Pull request by stevendanna**
Previously, we did not restore the system.role_options table when executing RESTORE SYSTEM USERS. As a result, a user's role options would not be restored.

For example, a user create with

   CREATE USER foo WITH NOLOGIN

would lose the NOLOGIN option.

Now, we restore these options.

Fixes: #94107

Release note (bug fix): Fixed a bug in which RESTORE SYSTEM USERS would fail to restore role options.

### CODE REVIEW

File: pkg/ccl/backupccl/restore_job.go [hunk 1]
```diff
@@ -2847,35 +2849,80 @@ func (r *restoreResumer) restoreSystemUsers(
...
-					return err
+		if hasSystemRoleOptionsTable(systemTables) {
+			selectNonExistentRoleOptions := "SELECT * FROM crdb_temp_system.role_options temp_ro WHERE " +
+				"NOT EXISTS (SELECT * FROM system.role_options ro WHERE temp_ro.username = ro.username AND temp_ro.option = ro.option)"
+			roleOptions, err := executor.QueryBuffered(ctx, "get-role-options", txn, selectNonExistentRoleOptions)
+			if err != nil {
+				return err
+			}
+
+			dstHasIDColumn := r.execCfg.Settings.Version.IsActive(ctx, clusterversion.V22_2RoleOptionsTableHasIDColumn)
```
Comment (andyyang890) [inline pkg/ccl/backupccl/restore_job.go:2885]: nit: maybe rename this to `roleOptionsHasIDColumn`?

**Review by andyyang890 — APPROVED** [review]
Thanks for fixing this!

**Review by msbutler — APPROVED** [review]

**Comment by stevendanna** [comment]
bors r=msbutler,andyyang890

**Comment by stevendanna** [comment]
bors retry

**Comment by rafiss** [comment]
Just wondering if there are plans to backport this?

**Pull request merged**